Exception with message "No identifier specified for entity" is thrown at startup of a Docker native image

### Expected Behavior

Application should startup correctly as when it was happening in the previous versions (before `3`)

### Actual Behaviour

Application throws the following exception
```
ERROR i.m.c.h.jpa.EntityManagerFactoryBean - Hibernate mapping error
org.hibernate.AnnotationException: No identifier specified for entity: com.mycompany.micronautjpamysql.model.Book
	at org.hibernate.cfg.InheritanceState.determineDefaultAccessType(InheritanceState.java:266)
	at org.hibernate.cfg.InheritanceState.getElementsToProcess(InheritanceState.java:211)
	at org.hibernate.cfg.AnnotationBinder.bindClass(AnnotationBinder.java:783)
	at org.hibernate.boot.model.source.internal.annotations.AnnotationMetadataSourceProcessorImpl.processEntityHierarchies(AnnotationMetadataSourceProcessorImpl.java:225)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess$1.processEntityHierarchies(MetadataBuildingProcess.java:239)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:282)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.build(MetadataBuildingProcess.java:86)
	at org.hibernate.boot.internal.MetadataBuilderImpl.build(MetadataBuilderImpl.java:479)
	at org.hibernate.boot.internal.MetadataBuilderImpl.build(MetadataBuilderImpl.java:85)
	at org.hibernate.boot.MetadataSources.buildMetadata(MetadataSources.java:202)
	at io.micronaut.configuration.hibernate.jpa.EntityManagerFactoryBean.hibernateSessionFactoryBuilder(EntityManagerFactoryBean.java:188)
	at io.micronaut.configuration.hibernate.jpa.$EntityManagerFactoryBean$HibernateSessionFactoryBuilder2Definition.build(Unknown Source)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:136)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2238)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3193)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3179)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2723)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2685)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1555)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1052)
	at io.micronaut.configuration.hibernate.jpa.$EntityManagerFactoryBean$HibernateSessionFactory3Definition.build(Unknown Source)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:136)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2238)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3193)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3179)
	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:2555)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1834)
	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:235)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3362)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:243)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:181)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:71)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:311)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:297)
	at com.mycompany.micronautjpamysql.Application.main(Application.java:8)
ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Bean definition [org.hibernate.SessionFactory] could not be loaded: Error instantiating bean of type  [org.hibernate.SessionFactory]

Message: No identifier specified for entity: com.mycompany.micronautjpamysql.model.Book
Path Taken: SessionFactory.hibernateSessionFactory([SessionFactoryBuilder sessionFactoryBuilder])
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [org.hibernate.SessionFactory] could not be loaded: Error instantiating bean of type  [org.hibernate.SessionFactory]

Message: No identifier specified for entity: com.mycompany.micronautjpamysql.model.Book
Path Taken: SessionFactory.hibernateSessionFactory([SessionFactoryBuilder sessionFactoryBuilder])
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1840)
	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:235)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3362)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:243)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:181)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:71)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:311)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:297)
	at com.mycompany.micronautjpamysql.Application.main(Application.java:8)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [org.hibernate.SessionFactory]

Message: No identifier specified for entity: com.mycompany.micronautjpamysql.model.Book
Path Taken: SessionFactory.hibernateSessionFactory([SessionFactoryBuilder sessionFactoryBuilder])
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2265)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3193)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3179)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2723)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2685)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1555)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1052)
	at io.micronaut.configuration.hibernate.jpa.$EntityManagerFactoryBean$HibernateSessionFactory3Definition.build(Unknown Source)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:136)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2238)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3193)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3179)
	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:2555)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1834)
	... 8 common frames omitted
Caused by: org.hibernate.AnnotationException: No identifier specified for entity: com.mycompany.micronautjpamysql.model.Book
	at org.hibernate.cfg.InheritanceState.determineDefaultAccessType(InheritanceState.java:266)
	at org.hibernate.cfg.InheritanceState.getElementsToProcess(InheritanceState.java:211)
	at org.hibernate.cfg.AnnotationBinder.bindClass(AnnotationBinder.java:783)
	at org.hibernate.boot.model.source.internal.annotations.AnnotationMetadataSourceProcessorImpl.processEntityHierarchies(AnnotationMetadataSourceProcessorImpl.java:225)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess$1.processEntityHierarchies(MetadataBuildingProcess.java:239)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:282)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.build(MetadataBuildingProcess.java:86)
	at org.hibernate.boot.internal.MetadataBuilderImpl.build(MetadataBuilderImpl.java:479)
	at org.hibernate.boot.internal.MetadataBuilderImpl.build(MetadataBuilderImpl.java:85)
	at org.hibernate.boot.MetadataSources.buildMetadata(MetadataSources.java:202)
	at io.micronaut.configuration.hibernate.jpa.EntityManagerFactoryBean.hibernateSessionFactoryBuilder(EntityManagerFactoryBean.java:188)
	at io.micronaut.configuration.hibernate.jpa.$EntityManagerFactoryBean$HibernateSessionFactoryBuilder2Definition.build(Unknown Source)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:136)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2238)
	... 21 common frames omitted
```

### Steps To Reproduce

- git clone https://github.com/ivangfr/graalvm-quarkus-micronaut-springboot.git
- cd graalvm-quarkus-micronaut-springboot/jpa-mysql
- docker-compose up -d
- cd micronaut-jpa-mysql
- ./mvnw clean package -Pgraalvm -Dpackaging=docker-native -Djib.to.image=ivanfranchin/micronaut-jpa-mysql-native:1.0.0
- run the docker native container
  ```
  docker run --rm --name micronaut-jpa-mysql-native \
    -p 9089:8080 -e MICRONAUT_ENVIRONMENTS=native -e MYSQL_HOST=mysql \
    --network jpa-mysql_default \
    ivanfranchin/micronaut-jpa-mysql-native:1.0.0
  ```

### Environment Information

- Mac OS X
- Java 11

### Example Application

https://github.com/ivangfr/graalvm-quarkus-micronaut-springboot/tree/master/jpa-mysql/micronaut-jpa-mysql

### Version

3.0.1

--- a/starter-core/src/main/java/io/micronaut/starter/build/maven/MavenBuildCreator.java
+++ b/starter-core/src/main/java/io/micronaut/starter/build/maven/MavenBuildCreator.java
@@ -71,14 +71,21 @@ public class MavenBuildCreator {
                 .artifactId("micronaut-validation")
                 .versionProperty("micronaut.version")
                 .buildCoordinate(true);
+        Coordinate mnGraal = Dependency.builder()
+                .groupId("io.micronaut")
+                .artifactId("micronaut-graal")
+                .versionProperty("micronaut.version")
+                .buildCoordinate(true);
 
         if (combineAttribute == MavenCombineAttribute.OVERRIDE) {
             annotationProcessorsCoordinates.add(injectJava);
             annotationProcessorsCoordinates.add(validation);
+            annotationProcessorsCoordinates.add(mnGraal);
         }
         if (testCombineAttribute == MavenCombineAttribute.OVERRIDE) {
             testAnnotationProcessorsCoordinates.add(injectJava);
             testAnnotationProcessorsCoordinates.add(validation);
+            annotationProcessorsCoordinates.add(mnGraal);
         }
 
         annotationProcessorsCoordinates.sort(Coordinate.COMPARATOR);

--- a/starter-core/src/test/groovy/io/micronaut/starter/feature/other/ProjectLombokSpec.groovy
+++ b/starter-core/src/test/groovy/io/micronaut/starter/feature/other/ProjectLombokSpec.groovy
@@ -87,4 +87,20 @@ class ProjectLombokSpec extends ApplicationContextSpec implements CommandOutputF
     </dependency>
 """)
     }
+
+    void 'micronaut-graal dependency is added because it is in the parent pom'() {
+        when:
+        String template = new BuildBuilder(beanContext, BuildTool.MAVEN)
+            .features(['lombok'])
+            .render()
+
+        then:
+        template.contains('''
+            <path>
+              <groupId>io.micronaut</groupId>
+              <artifactId>micronaut-graal</artifactId>
+              <version>${micronaut.version}</version>
+            </path>
+''')
+    }
 }

